v2.go

```diff
@@ -157,38 +154,43 @@ func (header *Header) formatVersion2() ([]byte, error) {
 			return nil, err
 		}
 		buf.Write(hdrLen)
-		addrSrc = header.SourceAddress.To4()
-		addrDst = header.DestinationAddress.To4()
+		sourceIP, destIP, _ := header.IPs()
+		addrSrc = sourceIP.To4()
+		addrDst = destIP.To4()
 	} else if header.TransportProtocol.IsIPv6() {
 		hdrLen, err := addTLVLen(lengthV6Bytes, len(header.rawTLVs))
 		if err != nil {
 			return nil, err
 		}
 		buf.Write(hdrLen)
-		addrSrc = header.SourceAddress.To16()
-		addrDst = header.DestinationAddress.To16()
+		sourceIP, destIP, _ := header.IPs()
+		addrSrc = sourceIP.To16()
+		addrDst = destIP.To16()
 	} else if header.TransportProtocol.IsUnix() {
 		buf.Write(lengthUnixBytes)
-		// TODO is below right?
-		addrSrc = []byte(header.SourceAddress.String())
-		addrDst = []byte(header.DestinationAddress.String())
+		sourceAddr, destAddr, ok := header.UnixAddrs()
+		if !ok {
+			return nil, ErrInvalidAddress
+		}
+		addrSrc = []byte(sourceAddr.Name)
+		addrDst = []byte(destAddr.Name)
+	}
+	if addrSrc == nil || addrDst == nil {
+		return nil, ErrInvalidAddress
 	}
 	buf.Write(addrSrc)
 	buf.Write(addrDst)
 
-	portSrcBytes := func() []byte {
-		a := make([]byte, 2)
-		binary.BigEndian.PutUint16(a, header.SourcePort)
-		return a
-	}()
-	buf.Write(portSrcBytes)
+	if sourcePort, destPort, ok := header.Ports(); ok {
+		portBytes := make([]byte, 2)
```

Comment:
Shouldn't it be 4 instead?

Reply:
The part is an `uint16`, so 2 bytes, unless I'm missing something? `portBytes` is overwritten for each port (source, then dest).

I suppose we could instead put the source port in `portBytes[0:2]` and the destination port in `portBytes[2:4]`, then write the whole thing.

Reply:
Nevermind, I had misread the code and it's fine to overwrite as there's a write to the buffer before that.